src/Mod/Draft/draftfunctions/move.py

```diff
@@ -215,4 +215,116 @@ def copy_moved_edge(object, edge_index, vector):
         vertex2 = object.Placement.multVec(object.Points[edge_index+1]).add(vector)
     return make_line.make_line(vertex1, vertex2)
 
-## @}
+
+# This part is higly experimental and used for moveSubElements function
+
+def parse_shape(shape, selected_vertexes, vector):
+    """ Parse the given shape and rebuild it according to its
+    original topological structure
+    """
+    import Part
+    if shape.ShapeType == "Compound":
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        compound = Part.Compound(new_sub_shapes)
+        return compound
+
+    elif shape.ShapeType == "CompSolid":
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        compound = Part.makeCompound(new_sub_shapes)
+
+        return shape
+
+    elif shape.ShapeType == "Solid":
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        solid = Part.Solid(new_sub_shapes)
+        return solid
+
+    elif shape.ShapeType == "Shell":
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        shell = Part.Shell(new_sub_shapes)
+        return shell
+
+    elif shape.ShapeType == "Face":
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        face = Part.Face(new_sub_shapes)
+        return face
+
+    elif shape.ShapeType == "Wire":
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
+        wire = Part.Wire(new_sub_shapes)
+        return wire
+
+    elif shape.ShapeType == "Edge":
+        # to be refined
+        new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)
```

Comment:
As far as I can see, the `parse_sub_shapes` function is called for all types of objects, except vertex. So maybe this should be placed outside the `if`. The idea is to avoid writing the same code multiple times.
```py
if shape.ShapeType in ("Compound", "CompSolid", "Solid", ...):
    new_sub_shapes = parse_sub_shapes(shape, selected_vertexes, vector)

if shape.ShapeType == "Compound":
    shape = Part.Compound(new_sub_shapes)
elif shape.ShapeType == "CompSolid":
    shape = Part.makeCompound(new_sub_shapes)
...

return shape
```

Also, storing the result of the `Part.` functions and classes `shape`, we don't need to place multiple return values. We can just use the last statement of the function, which is `return shape`, and it will return whatever was set up by the `if`s.